psi4/src/psi4/libmints/writer.cc

```diff
@@ -795,6 +799,9 @@ void NBOWriter::write(const std::string &filename) {
         }
         printer->Printf("%15.6E", exponents.get(0, i));
     }
+    // TODO: Rewrite to eliminate code duplication. The only thing changing between
```

Comment:
Rather than adding a TODO, I feel like it's not that much work to roll this up into a loop; who knows how long the C++ version of NBOWriter will remain around.

I suppose the iteration over the names can be done using something like:
```
char coeff_labels[] = { 'S', 'P', 'D', 'F', 'G', 'H' }; 
for (int am = 0; i < basisset.max_am(); am++) {
    ...
    printer->Printf("\n      C%s =", coeff_labels[am]);
    ...
}
```